### ISSUE TITLE
aggregate fetchXml query over linked entity field working?

### ISSUE THREAD

**Issue by mpseidel** [jordimontana82/fake-xrm-easy#392]
Hey there,

first of all thanks a lot for building this - adds a ton of value.

I'm trying to run a fetchXml query in memory which aggregates over an attribute of a child entity. Works fine against the real CRM but in against the fake org service.

I'm getting the field with AliasedValues but they value is always `null` instead of the expected sum amount.

This is how my query looks like. 

```xml
<fetch version='1.0' output-format='xml-platform' mapping='logical' distinct='true' aggregate='true'>
  <entity name='contact'>
	<attribute name='contactid' groupby='true' alias='agg_contactid' />
	<filter type='and'>
	  <condition attribute='my_somedate' operator='on-or-before' value='2019-02-01' />
	</filter>
	
	<link-entity name='my_sale' from='my_contact' to='contactid' alias='sales12m' link-type='inner'>
		<attribute name='my_amount' aggregate='sum' alias='TotalAmount' />
	</link-entity>
	
	
  </entity>
</fetch>
```

Should this be working currently?

**Comment by BetimBeja** [comment]
Hello @mpseidel ,
currently this is not working correctly and one reason is [this](https://github.com/jordimontana82/fake-xrm-easy/blob/cf08a13c5831380ada946e063d939ad380728f06/FakeXrmEasy.Shared/XrmFakedContext.Aggregations.cs#L35).
Also `my_somedate` is an attribute of `contact` or `my_sale`? If the answer is `my_sale` then you are missing a `entityname` attribute in the filter or should put the filter in the `link-entity`.
I will check if there is a quick solution for this...
Best regards,
Betim.

**Comment by mpseidel** [comment]
@BetimBeja thanks for the swift response! 
`my_somedate` would be an attribute of `contact` - I simplified the original fetchXml a bit - the original was working fine aggainst the real backend.

If there was a quick solution you would totally make my day :) but I assume it won't be that easy..

**Comment by BetimBeja** [comment]
@mpseidel , are you using early bound or late bound entities? Currently arithmetical FetchXml aggregations work only on early bound :( 
I will check about the solution probably today after work. 
@jordimontana82 has to review this though and my PRs are piling up 😄

**Comment by mpseidel** [comment]
@BetimBeja we use early bound types - fingers crossed :)

**Comment by mpseidel** [comment]
@BetimBeja I tried the code in your PR and it worked for my scenario!

I have one more query that is a bit more complex. It actually has a nested <link-entity> 

`sales_oder (recent sales oders) > account (for related accounts) > sales_order (calculate sum for account)`

With this query the aggregate sum valules keep returning `null`.

I wouldn't expect this to work but maybe its actually possible?

**Comment by mpseidel** [comment]
@BetimBeja actually your change would probably be enough - any change you can look at this one for approval @jordimontana82 ? :)

**Comment by jordimontana82** [comment]
That was merged in v1.52.0 so closing issue

**Comment by mpseidel** [comment]
Cool! Can we add support for max / min for DateTimes as well?

Getting an error `Message: System.Exception : Unhndled property type 'System.DateTime' in 'max' aggregate`
